sys/mips/signal.c

```diff
@@ -68,32 +69,40 @@ int sig_send(signo_t sig, sigaction_t *sa) {
 }
 
 int sig_return(void) {
+  int error = 0;
   thread_t *td = thread_self();
-  SCOPED_MTX_LOCK(&td->td_lock);
   sig_ctx_t ksc;
-  exc_frame_t *uframe = td->td_uframe;
```

Comment:
Please strive to minimize volume of code changes – that'll make reviews go faster. Especially avoid unnecessary code movement or indentation. I think current `sig_return` could made `static` and renamed to `copy_sigcontext`. While new `sig_return` would contain new code and call to `copy_sigcontext`.

Reply:
I find that such refactoring only makes it harder to follow the code when reading it -- I can understand it when it leads to code reuse. To minimize noise in the diff, you can ignore whitespace changes by checking "hide whitespace changes" in the settings at the top of the page. Unfortunately, you need to set it manually every time, there's no way to make it the default.

Reply:
Ok. I took me a few seconds to find the option.